src/EventListener/DataContainer/ModuleDataContainerListener.php

```diff
@@ -64,14 +65,16 @@ public function getModules(): array
             ),
         );
 
+        // Get all modules from DB.
         $themeCollection = \ThemeModel::findAll(array('order' => 'name'));
         while ($themeCollection->next()) {
             $modules[$themeCollection->name] = array();
 
             $moduleCollection = \ModuleModel::findBy('pid', $themeCollection->id, array('order' => 'name'));
             if ($moduleCollection) {
                 while ($moduleCollection->next()) {
-                    $modules[$themeCollection->name][$moduleCollection->id] = $moduleCollection->name;
+                    $modules[$GLOBALS['TL_LANG']['merger2']['legend_module']
+                             . $themeCollection->name][$moduleCollection->id] = $moduleCollection->name;
```

Comment:
I don't like the concatenation done inside the key statement. Please make two statements for it to increase readability:

```php
$category           = sprintf($GLOBALS['TL_LANG']['merger2']['legend_module'], $moduleCollection->id);
$modules[$category] = $moduleCollection->name;
```

Reply:
changed